### PR TITLE
lsix: init at 1.7.4

### PR DESCRIPTION
**Pull request by kidonng**
###### Motivation for this change

Add [lsix](https://github.com/hackerb9/lsix)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130526` at 007fc4ec run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>lsix</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/graphics/lsix/default.nix:16:3:
```
   |
16 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/tools/graphics/lsix/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -d $out/bin
+    install lsix $out/bin
+    wrapProgram $out/bin/lsix --prefix PATH : ${lib.makeBinPath [ imagemagick ] }
```
Comment (musfay) [inline pkgs/tools/graphics/lsix/default.nix:21]:
```suggestion
    install -Dm755 lsix -t $out/bin
```

File: pkgs/tools/graphics/lsix/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    runHook postInstall
+  '';
+
```
Comment (musfay) [inline pkgs/tools/graphics/lsix/default.nix:25]:
```suggestion

  postFixup = ''
    wrapProgram $out/bin/lsix \
      --prefix PATH : ${lib.makeBinPath [ imagemagick ]}
  '';
 
```

File: pkgs/tools/graphics/lsix/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{ lib, stdenv, fetchFromGitHub, makeWrapper, imagemagick }:
```
Comment (musfay) [inline pkgs/tools/graphics/lsix/default.nix:1]:
```suggestion
{ lib, stdenvNoCC, fetchFromGitHub, makeWrapper, imagemagick }:
```

File: pkgs/tools/graphics/lsix/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+stdenv.mkDerivation rec {
```
Comment (musfay) [inline pkgs/tools/graphics/lsix/default.nix:3]:
```suggestion
stdenvNoCC.mkDerivation rec {
```

**Review by musfay — CHANGES_REQUESTED** [review]

File: pkgs/tools/graphics/lsix/default.nix (hunk 1 above)
Comment (kidonng) [reply pkgs/tools/graphics/lsix/default.nix:21]: Thanks! I didn't know `install` in coreutils has this `-t` flag.

File: pkgs/tools/graphics/lsix/default.nix (hunk 3 above)
Comment (kidonng) [reply pkgs/tools/graphics/lsix/default.nix:1]: I see [neofetch](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/misc/neofetch/default.nix) uses `stdenvNoCC` but [any-nix-shell](https://github.com/NixOS/nixpkgs/blob/master/pkgs/shells/any-nix-shell/default.nix) uses `stdenv` so I was not sure whether one I should use. I thought it was whether there is a `Makefile` or not.

**Pull request merged**